dz_sprint_13

powerrewop opened the pull request:
Спринт 13 исправления замечаний 2:
1) Цвет курсора исправил
2) Стрелку назад поправил на всех формах
3) цвет боковых кнопок в темной теме - поправил




/////////////////////////////////////////////////
Спринт 13 исправления замечаний:

1) Стрелку назад исправил, теперь полное соответствие макету.
2) Длинный текст теперь корректно отображается в полях наименования трека и альбома
3) для полей класса Track - artworkUrl100, collectionName теперь возможно пустое значение
4) Что касается цвета курсора - ни в текущем спринте ни в предыдущих не было никаких требований к цвету курсора (эта тема еще не разбиралась) 

Примечание:
Для того чтобы форма плейера "влазила" на более маленькие экраны мне пришлось:
1) Уменьшил вертикальные отступы
2) Добавил ScrollView
3) уменьшил размер картинки альбома

Наставники сказали что приоритет нее подогнать форму под корректное отображение на маленьких дисплеях + добавить ScrollView. Данные отклонения от макета были вынужденными.

powerrewop committed:
dz_sprint_13

DmitrijEgorow commented on app/src/main/java/com/practicum/playlistmaker/PlayerActivity.kt:
```diff
@@ -0,0 +1,87 @@
+
+class PlayerActivity : AppCompatActivity() {
+
+    private lateinit var iv_back: ImageView
+    private lateinit var iv_banner: ImageView
+    private lateinit var tv_trackName: TextView
+    private lateinit var tv_trackAlbum: TextView
```
Для наименования полей класса не используют подчеркивания

Подробнее про стиль можно прочитать в [документации](https://kotlinlang.org/docs/coding-conventions.html#naming-rules) или [здесь](https://developer.android.com/kotlin/style-guide#naming_2)

DmitrijEgorow commented on app/src/main/java/com/practicum/playlistmaker/PlayerActivity.kt:
```diff
@@ -0,0 +1,87 @@
+        iv_back.setOnClickListener{
+            finish()
+        }
+
+
+
```
Много пустых строк

DmitrijEgorow commented on app/src/main/java/com/practicum/playlistmaker/Track.kt:
```diff
@@ -8,5 +8,9 @@ data class Track(
     var artworkUrl100: String,
     var trackId: Long,
-    var isHistory: Boolean = false
+    var isHistory: Boolean = false,
+    var collectionName: String,
+    var releaseDate: String,
```
Лучше оставить в [data class только неизменяемые свойства](https://www.reddit.com/r/Kotlin/comments/qibnqc/comment/hijhupt/?utm_source=share&utm_medium=web3x&utm_name=web3xcss&utm_term=1&utm_content=share_button)

DmitrijEgorow commented on app/src/main/res/layout/activity_player.xml:
```diff
@@ -0,0 +1,291 @@
+            app:layout_constraintGuide_begin="16dp" />
+
+        <androidx.constraintlayout.widget.Guideline
+            android:id="@+id/rightBorder"
```
Разные стили именования идентификаторов, где-то через _
Стоит унифицировать

DmitrijEgorow commented on app/src/main/res/layout/activity_player.xml:
```diff
@@ -0,0 +1,291 @@
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            android:orientation="vertical"
+            app:layout_constraintGuide_end="24dp" />
```
Стоит вынести в dimens

DmitrijEgorow commented on app/src/main/java/com/practicum/playlistmaker/PlayerActivity.kt:
```diff
@@ -0,0 +1,87 @@
+        tv_trackCountryInfo = findViewById<TextView>(R.id.tv_track_country_data)
+
+        val arguments  = getIntent().getExtras();
+        val json = arguments?.getString("TrackData")
```
Стоит вынести в константу с говорящим названием

DmitrijEgorow commented:
[Нарушение принципа сокрытия данных, стоит сделать всё приватным](https://github.com/powerrewop/PlaylistMaker/blob/44b24aa1b5485e3dc67160a49be87258f825bbf9/app/src/main/java/com/practicum/playlistmaker/App.kt#L13)

DmitrijEgorow commented:
[Плохое название, непонятно о чём оно](https://github.com/powerrewop/PlaylistMaker/blob/44b24aa1b5485e3dc67160a49be87258f825bbf9/app/src/main/java/com/practicum/playlistmaker/App.kt#L7)

DmitrijEgorow commented on app/src/main/java/com/practicum/playlistmaker/PlayerActivity.kt:
```diff
@@ -0,0 +1,87 @@
+
+
+
+        Glide.with(this).load(track.artworkUrl100.replaceAfterLast('/',"512x512bb.jpg")).placeholder(R.drawable.empty_image)
```
Стоит вынести замену отдельно, она ещё нам понадобится

DmitrijEgorow commented:
- [ ] Название трека и альбом могут быть длинными
- [ ] Стрелка возврата не соответствует дизайну
![Screenshot_20240719_005041](https://github.com/user-attachments/assets/c8179c1d-f6fa-4275-ab52-86c4dbe39f7d)

DmitrijEgorow commented:
- [ ] Цвет курсора не соответствует дизайну, должен быть синим

https://github.com/user-attachments/assets/0656f90f-be4e-4757-a134-caef09303cd3

powerrewop committed:
dz_sprint_13 исправления:
1) Стрелку назад исправил, теперь полное соответсвие макету.
2) Длинный текст теперь корректно отображается в полях наименования трека и альбома
3) для полей класса Track artworkUrl100, collectionName теперь возможно пустое значение
4) Что касается цвета курсура - ни в текущем спринте ни в предыдущих небыло никаких требований к цвету курсора (эта тема еще не разбиралась)

Примечание:
Для того чтобы форма плейера "влазила" на более маленькие экраны мне пришлось:
1) Уменьшить отступы
2) Добавил ScrollView
3) уменьшил размер картинки альбома

Наставники сказали что приоритетнее подогнать форму под корректное отображение на маленьких дисплеях + добавить ScrollView. Данные отклонения от макета были вынужденными.

DmitrijEgorow commented:
> 4. Что касается цвета курсора - ни в текущем спринте ни в предыдущих не было никаких требований к цвету курсора (эта тема еще не разбиралась)


Есть требование про соответствие дизайн-макету, там специально нарисован курсор в обеих темах, нужно поправить. В 9-м спринте обсуждали стили
![Screenshot_20240719_181735](https://github.com/user-attachments/assets/bd7ca909-c1ba-4d22-8d79-bfb85e5222d4)

Стрелку нужно поправить во всех местах и в обеих темах, выглядит странно, что её размеры изменяются при переходе с экрана на экран

DmitrijEgorow commented:
❌ Цвет боковых кнопок в тёмной теме

![Screenshot_20240720_090734](https://github.com/user-attachments/assets/fde1082c-3772-47f1-b549-56a962579a62)

powerrewop committed:
dz_sprint_13 исправления 2:
1) Цвет курсора исправил
2) Стрелку назад поправил на всех формах
3) цвет боковых кнопок в темной теме - поправил
